src/source/signal_generator.rs

```diff
@@ -71,11 +71,21 @@ impl SignalGenerator {
     pub fn new(sample_rate: cpal::SampleRate, frequency: f32, f: Function) -> SignalGenerator {
         assert!(frequency != 0.0, "frequency must be greater than zero");
         let period = sample_rate.0 as f32 / frequency;
+        let delta_theta = 1.0f32 / period;
+
+        let function: fn(f32) -> f32 = match f {
+            Function::Sine => sine_signal,
+            Function::Triangle => triangle_signal,
+            Function::Square => square_signal,
+            Function::Sawtooth => sawtooth_signal,
+        };
+
         SignalGenerator {
             sample_rate,
+            function,
+            delta_theta,
```

Comment:
maybe rename:
- delta_theta to step_size
- theta to position

that might be clearer for those without a math background :)

Reply:
could call it relative_position too, that might help to communicate that it wraps around at 1.0

A doc comment in the SignalGenerator struct above the variable stating that is nice too.

Reply:
Counterproposal: what if I called them `phase` and `phase_step`?

Reply:
Sorry I really don't like "position" for this, to me it implies something spatial and unbounded, and people might think it's tantamount to a time cursor which it definitely isn't. I think anybody who wants to get into hacking this struct is going to be familiar enough with synthesis that they'll recognize `phase` for what it means, and this term preserves the sense that it's moving within the bounded domain.

Reply:
yeah thats nice! good idea!

Reply:
Fixed in 8130c5b4149bec94d90e3ff05c57a4810144a972